File: dialog/src/main/java/com/caldremch/dialog/action_sheet/IData.java
package com.caldremch.dialog.action_sheet;

/**
 * @author: Caldremch
 * @date: 2020/7/9
 * @describe: 不同的ActionSheet共用一个ActionData的时候,
 * <p>
 * {@link BaseActionSheetDialog#getData()} 返回的数据继承{@link IData}
 * 对于不同的点击事件传递的数据进行区分
 **/

public interface IData {
    int sheetType();
}

File: data-core/src/main/java/com/caldremch/data/IData.kt
package com.caldremch.data

/**
 *
 * @author Caldremch
 *
 * @date 2020-07-03 13:55
 *
 * @email caldremch@163.com
 *
 * @describe
 *
 **/
interface IData {
    fun put(key: String, value: String?): IData
    fun put(key: String, values: Set<String>): IData
    fun put(key: String, value: Int): IData
    fun put(key: String, value: Long): IData
    fun put(key: String, value: Float): IData
    fun put(key: String, value: Boolean): IData
    fun remove(key: String?): IData
    fun clear(): IData
}